src/Fmt_ast.ml

```diff
@@ -1713,16 +1713,28 @@ and fmt_expression c ?(box= true) ?epi ?eol ?parens ?ext ({ast= exp} as xexp)
         $ fmt_atrs )
   | Pexp_open (flag, {txt; loc}, e0) ->
       let override = Poly.(flag = Override) in
+      let let_open =
+        match c.conf.let_open with
+        | `Preserve ->
+            let filter = function Parser.IN -> true | _ -> false in
+            let from = loc and upto = e0.pexp_loc in
+            let toks = Source.tokens_between c.source ~from ~upto ~filter in
```

Comment:
What are the case where `from.loc_end` > `upto.loc_start` ?

Reply:
I think the failure I saw was on `Ast_mapper.{default_mapper with location}`.

Reply:
I get two errors:
EXCEPTION: Negative length: -10
File "src/Cmts.ml", line 189, characters 16-6
EXCEPTION: Negative length: -10
File "src/Cmts.ml", line 190, characters 25-15
The corresponding code is:
```
      Ast_mapper.{default_mapper with location}
      (map_ast Ast_mapper.{default_mapper with attribute} ast)
```

Reply:
In those cases, `loc` and `e0.pexp_loc` will/should start at the same position.
Could we not call `tokens_between` when that condition hold ?

FYI, in following code (in the ocaml parser) makes both Pexp_open and Pexp_record have the same location. 
```ocaml
  | mod_longident DOT LBRACE record_expr RBRACE
      { let (exten, fields) = $4 in
        let rec_exp = mkexp(Pexp_record(fields, exten)) in
        mkexp(Pexp_open(Fresh, mkrhs $1 1, rec_exp)) }
```